File: Track/5. Sorting/8. Count the number of possible triangles .py
class Solution:
    def findNumberOfTriangles(self, arr, n):
        if n < 3:
            return 0
        a = sorted(arr)
        count = 0
        for i in range(n - 1, 0, -1): 
            l = 0; 
            r = i - 1; 
            while(l < r): 
                if(a[l] + a[r] > a[i]): 
      
                    count += r - l;  
                    r -= 1;  
                  
                else:
                    l += 1;
        return count

File: Track/5. Sorting/8. Count the number of possible triangles .c
int findNumberOfTriangles(int arr[], int n){
       
        
       qsort(arr, n, sizeof(arr[0]), comp);
 
    
    int count = 0;
 
    
    for (int i = 0; i < n - 2; ++i) {
       
        int k = i + 2;
 
        
        for (int j = i + 1; j < n; ++j) {
          
            while (k < n && arr[i] + arr[j] > arr[k])
                ++k;
 
           
            if (k > j)
                count += k - j - 1;
        }
    }
 
    return count;
    }
